tower-buffer/src/lib.rs

```diff
@@ -27,16 +27,16 @@ use std::sync::atomic::Ordering;
 /// Adds a buffer in front of an inner service.
 ///
 /// See crate level documentation for more details.
-pub struct Buffer<T>
-where T: Service,
+pub struct Buffer<T, R>
+where T: Service<R>,
```

Comment:
This isn't needed anymore, is it? I think `Buffer` can be made generic over only `R`.